gears/src/application/handlers.rs

```diff
@@ -54,24 +55,30 @@ pub trait TxHandler {
             account_number: account.account.get_account_number(),
         };
 
-        let tx_body = TxBody {
-            messages: vec![msg],
-            memo: String::new(),                    // TODO: remove hard coded
-            timeout_height: 0,                      // TODO: remove hard coded
-            extension_options: vec![],              // TODO: remove hard coded
-            non_critical_extension_options: vec![], // TODO: remove hard coded
-        };
-
+        let fee = Fee::new_with_defaults(fee);
         let tip = None; //TODO: remove hard coded
 
-        let raw_tx = create_signed_transaction(vec![signing_info], tx_body, fee, tip, chain_id);
+        let signer_info: SignerInfo = SignerInfo {
+            public_key: Some(signing_info.key.get_gears_public_key()),
+            mode_info: ModeInfo::Single(SignMode::Direct),
+            sequence: signing_info.sequence,
+        };
+
+        let auth_info = AuthInfo::new(vec![signer_info], fee, tip);
+        let tx_body = TxBody::from(msg);
 
-        let client = HttpClient::new(tendermint::rpc::Url::try_from(node)?)?;
+        let raw_tx = create_signed_transaction(&[signing_info], auth_info, tx_body, chain_id);
 
-        broadcast_tx_commit(client, raw_tx)
+        commit_transaction(raw_tx, node)
     }
 }
 
+#[inline]
```

Comment:
Is it a good idea to be manually inlining functions? see https://matklad.github.io/2021/07/09/inline-in-rust.html

Reply:
The rule:
> Fourth, when building libraries, proactively add #[inline] to small non-generic functions. Pay special attention to impls: Deref, AsRef and the like often benefit from inlining. A library can’t anticipate all usages upfront, it makes sense to not prematurely pessimize future users. Note that #[inline] is not transitive: if a trivial public function calls a trivial private function, you need to #[inline] both. See [this benchmark](https://github.com/matklad/benchmarks/tree/91171269f0a6e260a27111d07661021a89d20085/rust-inline) for details.

is applicable for the case. I'm force inline for small functions. But you're right, it can be skipped.